src/System.Buffers/src/System/Buffers/IBufferPool.cs

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace System.Buffers
+{
+    public interface IBufferPool<T>
```

Comment:
What's the reason for this abstraction? And if we need an abstraction, why not abstract class?